eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Configuration/ConfigResolver.php

```diff
@@ -214,6 +224,33 @@ public function getParameter($paramName, $namespace = null, $scope = null)
         }
     }
 
+    /**
+     * Try to extract service name that asked for a parameter using debug_backtrace().
+     *
+     * @return string
+     */
+    private function extractServiceName()
+    {
+        foreach (debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 10) as $t) {
+            if (!isset($t['function']) || $t['function'] === 'getParameter' || $t['function'] === __FUNCTION__) {
+                continue;
+            }
+
+            // Extract service name from first service matching getXXService pattern
+            // We can only reverse engineer traditional service name, namspace is stripped from class name based services
+            if (\strpos($t['function'], 'get') === 0 && \strpos($t['function'], 'Service') === \strlen($t['function']) -7) {
+                $serviceName = \strtolower(\preg_replace('/\B([A-Z])/', '_$1', \str_replace('_', '.', \substr($t['function'], 3, -7))));
+                if ($this->container->has($serviceName)) {
```

Comment:
Given the most of services tend to be private now, I'm not sure this is worth the effort. For Symfony 3.4 it will log deprecated warning and for Symfony 4.x it will return false in these cases.

Reply:
I can instead match it against updateable and/or resettable service ids array as implied in description for detecting services where this is not an issue _(not a warning if updatable, but could still be deprecation notice worthy if we remove the feature in 3.x)_, to also avoid using container for here while at it.

Reply:
Fixed this, however I think I need to cross check against container as well as with the changes now I get the following outpout on EE 2.5:
```
08:32:32 WARNING   [app] ConfigResolver was used to load parameter "languages" before SiteAccess was loaded by the following services: "ezpublish.helper.language_resolver", "->getEzpublish_Api_InnerRepositoryService()". (...)

08:32:32 WARNING   [app] ConfigResolver was used to load parameter "repository" before SiteAccess was loaded by the following services: "->getEzpublish_Api_StorageEngineService()", "->getEzpublish_Spi_SearchService()", "->getEzpublish_Api_InnerRepositoryService()". (...)

08:32:32 WARNING   [app] ConfigResolver was used to load parameter "workflows" before SiteAccess was loaded by the following services: "->getWorkflowDefinitionMetadataRegistryService()", "->getWorkflowRegistryService()". (...)

08:32:32 WARNING   [app] ConfigResolver was used to load parameter "anonymous_user_id" before SiteAccess was loaded by the following services: "->getEzpublish_Api_InnerRepositoryService()". (...)
```

So it's no longer able to detect `getEzpublish_Api_InnerRepositoryService`, `getEzpublish_Api_StorageEngineService` and `getEzpublish_Spi_SearchService` service names unlike before (see JIRA).

Also remember this patch is mainly for current LTS releases, future releases should clean this up somewhat differently depending on how this feature is re-implemented on Symfony 4.

Reply:
Output with last changes:
```
10:25:43 WARNING   [app] ConfigResolver was used to load parameter "languages" before SiteAccess was loaded by services: "ezpublish.helper.language_resolver", "ezpublish.api.inner_repository". This can cause issues. Try to use ConfigResolver lazily, make commands that rely on them lazy loaded, or try to mark the service as lazy.

10:25:43 WARNING   [app] ConfigResolver was used to load parameter "repository" before SiteAccess was loaded by services: "ezpublish.api.storage_engine", "ezpublish.spi.search", "ezpublish.api.inner_repository". This can cause issues. Try to use ConfigResolver lazily, make commands that rely on them lazy loaded, or try to mark the service as lazy.

10:25:43 WARNING   [app] ConfigResolver was used to load parameter "workflows" before SiteAccess was loaded by services: "->getWorkflowDefinitionMetadataRegistryService()", "->getWorkflowRegistryService()". This can cause issues. Try to use ConfigResolver lazily, make commands that rely on them lazy loaded, or try to mark the service as lazy.

10:25:43 WARNING   [app] ConfigResolver was used to load parameter "anonymous_user_id" before SiteAccess was loaded by services: "ezpublish.api.inner_repository". This can cause issues. Try to use ConfigResolver lazily, make commands that rely on them lazy loaded, or try to mark the service as lazy.
```

Reply:
Fixes for issues in 1.7 proposed in https://github.com/ezsystems/ezpublish-kernel/pull/2649
Fixes for other branches will have to be detected one by one.